crates/utils/src/settings/mod.rs

```diff
@@ -81,15 +73,30 @@ impl Settings {
 
   pub fn save_config_file(data: &str) -> Result<String, LemmyError> {
     fs::write(CONFIG_FILE, data)?;
+    Ok(Self::read_config_file()?)
+  }
 
-    // Reload the new settings
-    // From https://stackoverflow.com/questions/29654927/how-do-i-assign-a-string-to-a-mutable-static-variable/47181804#47181804
-    let mut new_settings = SETTINGS.write().expect("write config");
-    *new_settings = match Settings::init() {
-      Ok(c) => c,
-      Err(e) => panic!("{}", e),
-    };
+  // TODO for these regexes: we can't use lazy_static here anymore, since the settings must be
+  // initialized first.
+  pub fn webfinger_community_regex(&self) -> Regex {
+    Regex::new(&format!("^group:([a-z0-9_]{{3,}})@{}$", self.hostname))
+      .expect("compile webfinger regex")
+  }
```

Comment:
This was extremely unfortunate, and I can't find a clean way to do this. Since settings is no longer a singleton, we can't use lazy static for regexes which are dependent on settings (specifically the hostname) being initialized. So its rebuilding this regex every single time.

I couldn't find a way to do it quickly, but there's a TODO in here.

Reply:
Probably need to add a member to the Settings struct like `Option<Regex>`, and only compile the regex here if it is none. And put `#[serde(skip)]`.

Reply:
That should be doable.